lib/chef/client.rb

```diff
@@ -637,7 +637,11 @@ def node_name
     # @api private
     #
     def register(client_name = node_name, config = Chef::Config)
-      if !config[:client_key]
+      if Chef::HTTP::Authenticator.detect_certificate_key(client_name)
```

Comment:
Should probably Chef::Log.warn here if the client_key exists on the filesystem but we're using the certificate store